### PR TITLE
[nG] Gather-8 shell

### PR DESCRIPTION
**Pull request by pavel-esir**
### Details:
 - the only difference from Gather-7 is that is supports negative indices
 - specification for Gather-8 https://github.com/openvinotoolkit/openvino/pull/6172
### Tickets:
 - 54625

### CODE REVIEW

File: ngraph/core/include/ngraph/op/gather.hpp [hunk 1]
```diff
@@ -61,5 +61,32 @@ namespace ngraph
...
+                /// \param data The tensor from which slices are gathered
+                /// \param indices Tensor with indexes to gather
+                /// \param axis The tensor is a dimension index to gather data from
+                /// \param batch_dims The number of batch dimension in data and indices tensors.
+                Gather(const Output<Node>& data,
+                       const Output<Node>& indices,
+                       const Output<Node>& axis,
+                       const int64_t batch_dims = 0);
+
+                bool visit_attributes(AttributeVisitor& visitor) override;
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/gather.hpp:83]: Could you add new test on `visit_attributes`?

**Review by sadolini — APPROVED** [review]

File: ngraph/core/include/ngraph/op/gather.hpp (hunk 1 above)
Comment (pavel-esir) [reply ngraph/core/include/ngraph/op/gather.hpp:83]: Added. Also bumped to opset8. Please review

File: ngraph/core/include/ngraph/opsets/opset8_tbl.hpp [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+NGRAPH_OP(Divide, ngraph::op::v1)
+NGRAPH_OP(Elu, ngraph::op::v0)
+NGRAPH_OP(Erf, ngraph::op::v0)
+NGRAPH_OP(Equal, ngraph::op::v1)
+NGRAPH_OP(Exp, ngraph::op::v0)
+NGRAPH_OP(ExtractImagePatches, ngraph::op::v3)
+NGRAPH_OP(FakeQuantize, ngraph::op::v0)
+NGRAPH_OP(Floor, ngraph::op::v0)
+NGRAPH_OP(FloorMod, ngraph::op::v1)
+NGRAPH_OP(Gather, ngraph::op::v8)
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/opsets/opset8_tbl.hpp:45]: Please add new operations for opset8 to the end of file.

File: ngraph/core/include/ngraph/op/gather.hpp [hunk 3]
```diff
@@ -61,5 +61,32 @@ namespace ngraph
                     clone_with_new_inputs(const OutputVector& new_args) const override;
             };
         } // namespace v7
+
+        namespace v8
```
Comment (ilyachur) [inline ngraph/core/include/ngraph/op/gather.hpp:65]: Do we need to update specification?

Comment (pavel-esir) [reply ngraph/core/include/ngraph/op/gather.hpp:65]: yes, here is the spec for G8
https://github.com/openvinotoolkit/openvino/pull/6172

File: ngraph/core/include/ngraph/opsets/opset8_tbl.hpp (hunk 2 above)
Comment (pavel-esir) [reply ngraph/core/include/ngraph/opsets/opset8_tbl.hpp:45]: done

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
@pavel-esir Please add Python wrapper in the separate PR.

**Pull request merged**